File: A/a007.py
def prime(n):
    if n == 2:
        return True
    if n % 2 == 0:
        return False

    i = 3
    while i * i <= n:
        if n % i == 0:
            return False
        i += 2
    return True

while True:
    try:
        x = int(input())
        if prime(x):
            print("質數")
        else:
            print("非質數")
    except EOFError:
        break

File: A/a007.cpp
#include <iostream>
#include <math.h>
using namespace std;

int prime(int n){
    if (n == 2){
        return 1;
    }
    if (n % 2 == 0){
        return 0;
    }
    int i = 3;
    while (pow(i, 2) <= n){
        if (n % i == 0){
            return 0;
        };
        i += 2;
    }
    return 1;
}

int main(){
    int x;
    while (cin >> x){
        if (prime(x)){
            cout << "質數\n";
        }
        else{
            cout << "非質數\n";
        }
    }
    return 0;
}